Add beginning of software forest chapter

d33bs opened the pull request:
<!-- _modified from [EmbeddedArtistry](https://embeddedartistry.com/blog/2017/08/04/a-github-pull-request-template-for-your-projects/)_
_referenced with modifications from [pycytominer](https://github.com/cytomining/pycytominer/blob/master/.github/PULL_REQUEST_TEMPLATE.md)_ -->

# Description

This PR adds the beginning of the software forest chapter as part of addressing #24 . It is based upon the work found in #33 to help with linting needs discovered within the the changes of that PR.

References #24

## What is the nature of your change?

- [x] Content additions or updates (adds or updates content)
- [ ] Bug fix (fixes an issue).
- [ ] Enhancement (adds functionality).
- [ ] Breaking change (these changes would cause existing functionality to not work as expected).

# Checklist

Please ensure that all boxes are checked before indicating that this pull request is ready for review.

- [x] I have read the [CONTRIBUTING.md](CONTRIBUTING.md) guidelines.
- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own contributions.
- [ ] I have commented my content, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to related documentation (outside of book content).
- [x] My changes generate no new warnings.
- [x] New and existing tests pass locally with my changes.
- [ ] I have added tests that prove my additions are effective or that my feature works.
- [x] I have deleted all non-relevant text in this pull request template.

gwaybio commented on src/almanac/garden-lattice/garden-lattice.md:
```diff
@@ -0,0 +1,32 @@
+# Garden Lattice
```
Looks like this duplicates #33

d33bs replied:
Thanks, I'll update this PR before merge with changes from #33.

gwaybio commented on src/almanac/software-forest/software-forest.md:
```diff
@@ -0,0 +1,30 @@
+
+> "Called Three-Dimensional Forestry or Forest Farming, … the three 'dimensions' of \[Toyohiko Kagawa's\] 3-D system were the trees as conservers of the soil and suppliers of food and the livestock which benefited from them." {cite:p}`hart_forest_1996`, {cite:p}`kagawa_fujisaki_3d_farming`
+
+Forest gardening offers innovations from an ecological cultivation perspective by caring for trees, understory vegetation, soil, and other  environmental factors together.
```
```suggestion
Forest gardening offers innovations from an ecological cultivation perspective by caring for trees, understory vegetation, soil, and other environmental factors.
```

d33bs replied:
Thanks for this!

gwaybio commented on src/almanac/software-forest/software-forest.md:
```diff
@@ -0,0 +1,30 @@
+> "Called Three-Dimensional Forestry or Forest Farming, … the three 'dimensions' of \[Toyohiko Kagawa's\] 3-D system were the trees as conservers of the soil and suppliers of food and the livestock which benefited from them." {cite:p}`hart_forest_1996`, {cite:p}`kagawa_fujisaki_3d_farming`
+
+Forest gardening offers innovations from an ecological cultivation perspective by caring for trees, understory vegetation, soil, and other  environmental factors together.
+The software forest is a multi-layered forest garden enriched through connections and minimal, iterable change.
```
```suggestion
```

Not sure this sentence adds anything you don't already describe in the surrounding sentences

d33bs replied:
Thanks for this, I repurposed this with a more gradual introduction paragraph.

gwaybio commented on src/almanac/software-forest/software-forest.md:
```diff
@@ -0,0 +1,30 @@
+Others are exhibited through abstract relationships as [_nurse logs_](https://en.wikipedia.org/wiki/Nurse_log), [_Hügelkultur_](https://en.wikipedia.org/wiki/H%C3%BCgelkultur), or [mycorrhizal](https://en.wikipedia.org/wiki/Mycorrhiza)-like associations which redistribute energy among the entirety of a software region.
+
+As software gardeners we must consider how energy is transferred to various portions of the forest.
+Adventitious code, which "volunteers" by accident of growth, may create scenarios of energy waste (also ) in the form of cognitive or performance misalignment.
```
Not sure what is going on with the `(also )`?

d33bs replied:
Thanks, I'll remove.

gwaybio commented on src/almanac/software-forest/software-forest.md:
```diff
@@ -0,0 +1,30 @@
+# Software Forest
```
I'm a bit confused why we are introducing the software forest metaphor - is it a vehicle for us to describe software environments? Is it to discuss how software can evolve with minimal interference?

Where in the book does will this content go? If it is early, then it might risk overextending the software garden metaphor. I think it's better suited for later, after we've more fully described the garden.

d33bs replied:
Thanks for this comment. I found inspiration in forest gardens and thought the metaphors might extend well to explain many aspects of software, including how multidimensional it can be and how slow growth can be sustainable long-term. I'll create an issue to address this moving forward, because I feel you're right in that we should be specific with how this builds the experience within the Almanac.

d33bs replied:
Created #36

gwaybio reviewed (approved):
Looks like there are overlaps with #33 , so I recommend merging this first

d33bs commented:
Thanks @gwaybio ! I've addressed your comments and will now merge. I'm planning to revisit this content later as other material develops through #36 .
